### PR TITLE
Template Duplication

### PR DESCRIPTION
**Pull request by Daniel-Jang**
Original Issue: https://github.com/cass-degrees/CASS-Degrees-Code/issues/18

This allows the user to duplicate an existing degree/program that a user has selected from list.html for the user to alter it and create.

### CODE REVIEW

File: cassdegrees/ui/views/programs.py [hunk 1]
```diff
@@ -14,7 +30,10 @@ def create_program(request):
             return redirect('/list/?view=Degree&msg=Successfully Added Program!')
 
     else:
-        form = EditProgramFormSnippet()
+        if duplicate:
```
Comment (j-selby) [inline cassdegrees/ui/views/programs.py:35]: Wait, this will edit, not duplicate. You need to reset the ID, no?

**Review by j-selby — CHANGES_REQUESTED** [review]

File: cassdegrees/templates/list.html [hunk 2]
```diff
@@ -66,6 +66,10 @@
                                     {% else %}
                                         <td><p class="text-center"><input title="Select" type="checkbox" name="id" value="{{ item }}"  onchange="sessionStorage.setItem('selected', {{ item }})"/></p></td>
                                         <td>
+                                            {# Only Degree and Subplans will need this feature #}
+                                            {% if title == 'Degree' or title == 'Subplan'  %}
+                                                <a class="btn-uni-grad" href="/create/{% if title == 'Degree' %}program{% elif title == 'Subplan' %}subplan{% elif title == 'Course' %}course{% endif %}/?id={{ item }}&duplicate=True">Duplicate</a>
```
Comment (j-selby) [inline cassdegrees/templates/list.html]: Web syntax is generally lower-case false here.

File: cassdegrees/ui/views/programs.py [hunk 3]
```diff
@@ -6,6 +6,22 @@
 
 
 def create_program(request):
+    duplicate = request.GET.get('duplicate', False)
```
Comment (j-selby) [inline cassdegrees/ui/views/programs.py]: You are either returning a string or a boolean value here - little inconsistent.

File: cassdegrees/ui/views/programs.py (hunk 1 above)
Comment (Daniel-Jang) [reply cassdegrees/ui/views/programs.py:35]: No, I tried it out and it did not edit the existing one. That part of code is for GET requests, and that instance=instance was done just to display the contents of the program user selected to duplicate from.

I have left the POST part of create_program view function as it was before so adding is normal. Attempting to add a duplicate without changing anything will lead to this anyways:
image

File: cassdegrees/ui/views/programs.py (hunk 3 above)
Comment (Daniel-Jang) [reply cassdegrees/ui/views/programs.py]: Fixed it by setting it to 'false' string and then reassign duplicate variable as boolean True or False immediately after receiving it.

**Review by JackSekoranja — APPROVED** [review]
Looks good, works well. No complaints

**Review by KavsHub — APPROVED** [review]
This is pretty much the same as the subplan one. Happy for you to merge both of them in.

**Pull request merged**